### PR TITLE
Implement negative step for pixelbuf slices

### PR DESCRIPTION
**Pull request by dunkmann00**
@jepler mentioned this would be a useful improvement in #2936 

Tested on an itsybitsy m4 express

### CODE REVIEW

**Comment by jepler** [comment]
@dunkmann00 Thank you, that was quick!

This is a bit more complicated, I feel like I should actually test it.  But don't let me stop any other reviewer from giving it a go.

File: shared-bindings/_pixelbuf/PixelBuf.c [hunk 1]
```diff
@@ -300,17 +300,20 @@ STATIC mp_obj_t pixelbuf_pixelbuf_subscr(mp_obj_t self_in, mp_obj_t index_in, mp
         size_t length = common_hal__pixelbuf_pixelbuf_get_len(self_in);
         mp_seq_get_fast_slice_indexes(length, index_in, &slice);
 
-        if (slice.step < 0) {
-            mp_raise_IndexError(translate("Negative step not supported"));
+        size_t slice_len;
+        if (slice.step > 0) {
+            slice_len = slice.stop - slice.start;
+        }else{
```
Comment (jepler) [inline shared-bindings/_pixelbuf/PixelBuf.c:306]: Our coding style wants spaces: `} else {`.

File: shared-module/_pixelbuf/PixelBuf.c [hunk 2]
```diff
@@ -216,12 +216,19 @@ void _pixelbuf_set_pixel(pixelbuf_pixelbuf_obj_t* self, size_t index, mp_obj_t v
...
     size_t source_i = 0;
-    for (size_t target_i = start; target_i < stop; target_i += step) {
-        _pixelbuf_set_pixel(self, target_i, values[source_i]);
-        source_i++;
+    if (step > 0) {
+        for (size_t target_i = start; target_i < stop; target_i += step) {
+            _pixelbuf_set_pixel(self, target_i, values[source_i]);
+            source_i++;
+        }
+    }else{
```
Comment (jepler) [inline shared-module/_pixelbuf/PixelBuf.c:227]: Spaces around `else` please.

**Comment by jepler** [comment]
Here's the program, make sure to rename it to "pbs.py", getting rid of the ".txt" extension!  (GUI tools can make this tricky)

[pbs.py.txt](https://github.com/adafruit/circuitpython/files/4668632/pbs.py.txt)

**Comment by dunkmann00** [comment]
@jepler I started to look into this, but I think your test program fails in some scenarios if you try to set the list as well.

I just changed line 29 from:
`a[start:end:stride] = x`
to:
`l[start:end:stride] = x`

and I'm getting a `NotImplementedError: `

I'll try to understand this better, but what do you think about the list error?

**Comment by dunkmann00** [comment]
Ok, I fixed the problem with setting slices in pixebuf. It should work now.

**Comment by dunkmann00** [comment]
Okay, and should I file an issue about setting lists with slices where the step is not 1?

I might not have been clear in my responses, but that is a separate issue that I did not fix.

**Comment by jepler** [comment]
No, I think it's deliberately missing on lists. I wouldn't recommend filing an issue.

**Comment by dunkmann00** [comment]
Okay, sounds good. And sorry but I pushed another commit. This makes the set_pixels function slightly simpler.

**Pull request merged**